### PR TITLE
Move log setup into log module

### PR DESCRIPTION
**Pull request by kalikiana**
This is the additional step to stop requiring Setup to use logs.

Fixes: https://progress.opensuse.org/issues/27868

### CODE REVIEW

File: lib/OpenQA/Scheduler.pm [hunk 1]
```diff
@@ -18,7 +18,7 @@ use Mojo::Base 'Mojolicious';
 
 use OpenQA::Setup;
 use Mojo::IOLoop;
-use OpenQA::Log 'log_debug';
+use OpenQA::Log qw/log_debug setup_log/;
```
Comment (okurz) [inline lib/OpenQA/Scheduler.pm:21]: our common (or even required) style is `qw()` rather than `qw//`

File: t/24-worker-settings.t [hunk 2]
```diff
@@ -61,8 +61,8 @@ is_deeply(
 
 subtest 'apply settings to app' => sub {
     my ($setup_log_called, $setup_log_app);
-    my $mock = Test::MockModule->new('OpenQA::Setup');
-    $mock->mock(
+    my $mock = Test::MockModule->new('OpenQA::Worker::Settings');
```
Comment (Martchus) [inline t/24-worker-settings.t:64]: Shouldn't this mock the `OpenQA::Log` module now?

Comment (kalikiana) [reply t/24-worker-settings.t:64]: The import is `OpenQA::Log 'setup_log'` which means the function is logically in the importing module which is Settings.pm.

Comment (Martchus) [reply t/24-worker-settings.t:64]: Ok, considering that the tests pass you're likely right.

**Pull request merged**